app/src/main/java/com/compscieddy/meetinthemiddle/AuthenticationActivity.java

```diff
@@ -34,15 +41,24 @@ protected void onCreate(Bundle savedInstanceState) {
       finish();
     }
 
-    startActivityForResult(
-        AuthUI.getInstance().createSignInIntentBuilder()
-            .setLogo(R.mipmap.ic_launcher)
-            .setProviders(AuthUI.EMAIL_PROVIDER,
-                AuthUI.FACEBOOK_PROVIDER,
-                AuthUI.GOOGLE_PROVIDER)
-            .build(),
-        RC_SIGN_IN);
-    finish();
+    //User isn't logged in, so check if he has working internet
+    if (isOnline()) {
+      startActivityForResult(
+          AuthUI.getInstance().createSignInIntentBuilder()
+              .setLogo(R.mipmap.ic_launcher)
+              .setProviders(AuthUI.EMAIL_PROVIDER,
+                  AuthUI.FACEBOOK_PROVIDER,
+                  AuthUI.GOOGLE_PROVIDER)
+              .build(),
+          RC_SIGN_IN);
+      finish();
+    } else {
+      FragmentManager fm = getSupportFragmentManager();
+      InternetErrorFragment internetErrorFragment = InternetErrorFragment.newInstance();
+      internetErrorFragment.setStyle(DialogFragment.STYLE_NORMAL, R.style.AppDialogTheme);
+      internetErrorFragment.setCancelable(false);
+      internetErrorFragment.show(fm, "fragment_internet_error");
```

Comment:
I know it's not used now, but good practice to move out that string to a constant - also makes it more apparent that that's a fragment tag